blockchain/blockchain.go

```diff
@@ -672,9 +641,8 @@ func (bc *blockchain) MintNewDummyBlock() *Block {
 
 //  CommitBlock validates and appends a block to the chain
 func (bc *blockchain) CommitBlock(blk *Block) error {
-	if err := bc.ValidateBlock(blk); err != nil {
```

Comment:
i think it's OK to leave the code like this? Commit() = Validate(), then followed by private commit(), the intention/logic is quite clear

Reply:
I intentionally split upper case Commit/Validate and lower case commit/validate, where the former has lock protection. In general we should only lock on public methods, and I do this on purpose. All public methods should of blockchain (Start in this case) should call the lower case to prevent deadlock.